src/vs/editor/common/viewLayout/viewLineRenderer.ts

```diff
@@ -62,6 +62,7 @@ export class RenderLineInput {
 	public readonly renderWhitespace: RenderWhitespace;
 	public readonly renderControlCharacters: boolean;
 	public readonly fontLigatures: boolean;
+	public readonly lineHeight: number;
```

Comment:
Why is this field added if there are no usages?

Reply:
I finally remembered why I added the line height field. As mentioned above I added it to differentiate between two rendered line inputs through the `equals` method. I added this to rerender lines when the line height changes (please see viewLine.ts file line 169).

Reply:
I think this comes from the need to update rendered lines when their height change, but this is IMHO a bad way to achieve it. For example, rendered lines must also be updated when their top change (like when a view zone is inserted above them), yet the line top is not present as a field in this class. The idea of `RenderLineInput` is to describe all the data needed to render a line. We should not add fields because they will have side-effects somewhere, rather we should model the problem of updating the line height at the place where we set the line height.